storage_controller/src/service.rs

```diff
@@ -6374,7 +6387,13 @@ impl Service {
             }
         };
 
-        let units = match self.reconciler_concurrency.clone().try_acquire_owned() {
+        let acq = if high_priority {
+            self.prio_reconciler_concurrency.clone().try_acquire_owned()
```

Comment:
What if the priority reconciler semaphore is full but the normal isn't, then marking a request as priority is hurtful, no?

In order to make it truly prioritized, we could maybe have it first go to the priority reconciler semaphore, then to the normal one? also we'd probably need two queues or such.